ci_framework/playbooks/99-logs.yml

```diff
@@ -32,9 +32,15 @@
       ansible.builtin.import_role:
         name: artifacts
 
-    - name: Run os_must_gather
+    - name: Check if "success" flag exists
+      register: cifmw_success_flag
+      ansible.builtin.stat:
+        path: "{{ ansible_user_dir }}/cifmw-success"
+
+    - name: Run os_must_gather only when there is no cifmw-success flag or if it's forced
       ansible.builtin.import_role:
         name: os_must_gather
+      when: (cifmw_os_must_gather_run | default('false') | bool) or (not cifmw_success_flag.stat.exists)
```

Comment:
that's a good first approach. Just a tiny note in the doc about that new `cifmw_os_must_gather_run` global parameter and we should be OK.
Though.. moving that call (the whole import_role) in the artifacts role may help with:
- cifmw_artifacts_must_gather_run as an "artifacts parameter"
- running [artifacts/crc.yml](https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/artifacts/tasks/crc.yml) only if must_gather fails
- run that same artifacts/crc.yml only in case of failure as well

So checking for the flag would be in the artifacts role directly, for instance here:
https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/artifacts/tasks/main.yml#L46

with the import of os_must_gather role right after (with the conditional) and, why not, in a block like this:
```YAML
name: Run os_must_gather if needed, rescue with scp
block:
  - name: Run os_must_gather
     when: cifmw_artifacts_run_must_gather | bool or not cifmw_success_flag.stat.exists
     ansible.builtin.import_role:
      name: os_must_gather
rescue:
  - name: Gather CRC logs from disk
     ansible.builtin.import_tasks: crc.yml
```

What do you think?